src/main/java/io/cdap/plugin/gcp/gcs/sink/GCSBatchSink.java

```diff
@@ -82,18 +111,98 @@ public void prepareRun(BatchSinkContext context) throws Exception {
     if (bucket == null) {
       GCPUtils.createBucket(storage, config.getBucket(), config.getLocation(), cmekKey);
     }
+
   }
 
   @Override
   protected Map<String, String> getFileSystemProperties(BatchSinkContext context) {
-    return GCPUtils.getFileSystemProperties(config, config.getPath(), new HashMap<>());
+    return GCPUtils
+      .getFileSystemProperties(config, config.getPath(), new HashMap<>());
+  }
+
+  @Override
+  protected String getOutputDir(long logicalStartTime) {
+    this.outputPath =  super.getOutputDir(logicalStartTime);
```

Comment:
This method is overridden to store the value of outputPath. Since the path is appended with a timeSuffix, the exact output path is not known otherwise.

Reply:
Fixed the space.

Reply:
oh ok i see